subiquity/models/filesystem.py

```diff
@@ -867,14 +880,29 @@ class FilesystemModel(object):
             fs_by_name[fs.label] = fs
     fs_by_name['fat32'] = FS('fat32', True)
 
+    def _probe_bootloader(self):
```

Comment:
Consider using archdetect. It's on ubuntu-live images, probably could be on ubuntu-server-live images; it's meant to be used (package: archdetect-deb) to automatically gives you architecture and subarch (although it *is* coming from hw-detect, a d-i component; and probably best be extracted into it's own thing). You'd get a string like 'amd64/efi' or 'amd64/generic', that will give you the arch/subarch details that should be enough to tell exactly what bootloader you need; even to the point of telling whether you really need grub-ieee1374 or grub-whatever-else-for-ppc64el, + whatever else you might need for future support of arm/arm64.

Reply:
Well. We support a lot less than what hw-detect can do, as we target whatever curtin can do. Going forward we will probably have "uefi+bios", "uefi", "prep", "none". And I wonder if we should simply "patch" platform.machine() from cmdline for testing "different" things. Not relying on hw-detect imho is currently an advantage.